core/src/main/java/org/apache/calcite/sql/SqlCall.java

```diff
@@ -181,6 +181,10 @@ public int operandCount() {
     if (!this.getOperator().getName().equalsIgnoreCase(that.getOperator().getName())) {
       return litmus.fail("{} != {}", this, node);
     }
+    // For same name but different kind
+    if (this.getOperator().getKind() != that.getOperator().getKind()) {
```

Comment:
This looks wrong to me, shouldn't this be `return false`?
Doesn't this throw?

Reply:
@mihaibudiu You mean the test case i write should `return false`? It will `return true` for now. cause `UNARY_MINUS` and `MINUS` , their names are same, `-`. But It should probably check the operator.  I'll think about this change again

Reply:
@jiefei30 
If you could provide an additional method, 
it might be better than modifying the public method.

Reply:
@JiajunBernoulli There are more test cases i need to add.I'll add them these days.